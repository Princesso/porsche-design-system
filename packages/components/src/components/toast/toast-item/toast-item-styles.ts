import type { Theme } from '../../../types';
import type { ToastState } from '../toast/toast-utils';
import { getCss } from '../../../utils';
import {
  getCloseIconJssStyle,
  getNotificationContentJssStyle,
  getNotificationIconJssStyle,
  getNotificationRootJssStyle,
} from '../../inline-notification/inline-notification-styles';
import { getBoxShadow } from '../../banner/banner-styles';
import { textSmallStyle } from '@porsche-design-system/utilities-v2';
import { getThemedColors } from '../../../styles';

export const getComponentCss = (state: ToastState, theme: Theme): string => {
  const textColor = getThemedColors(theme).primaryColor;
  return getCss({
    '@global': {
      ':host': {
        ...getNotificationRootJssStyle(state, theme),
        ...getBoxShadow(),
      },
      p: {
        ...textSmallStyle,
        margin: 0,
        color: textColor,
      },
    },
    icon: getNotificationIconJssStyle(theme, state),
    content: getNotificationContentJssStyle(),
    close: getCloseIconJssStyle(),
  });
};
