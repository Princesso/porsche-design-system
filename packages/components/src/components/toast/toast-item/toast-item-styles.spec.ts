import { getComponentCss } from './toast-item-styles';

xdescribe('getComponentCss()', () => {
  it.each<Parameters<typeof getComponentCss>>([
    ['success', 'light'],
    ['info', 'light'],
    ['success', 'dark'],
    ['info', 'dark'],
  ])('should return correct css for state: %s and theme: %s', (...args) => {
    expect(getComponentCss(...args)).toMatchSnapshot();
  });
});
