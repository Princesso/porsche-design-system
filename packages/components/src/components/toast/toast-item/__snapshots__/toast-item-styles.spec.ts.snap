// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getComponentCss() should return correct css for state: info and theme: dark 1`] = `
":host {
  display: grid;
  grid-template-columns: 1fr auto;
  grid-template-rows: auto;
  grid-row-gap: 1rem;
  align-items: start;
  justify-items: start;
  padding: 1rem;
  background: #001A33;
  border-radius: 4px;
  box-shadow: 0 0.125rem 0.25rem 0 rgba(0,0,0,0.05),0 0.9375rem 1.25rem 0 rgba(0,0,0,0.2);
}
p {
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  margin: 0;
  color: #FBFCFF;
}
@media(min-width:760px) {
  :host {
    grid-template-columns: 2rem 1fr auto auto;
  }
}

.icon {
  display: none;
}
.content {
  display: grid;
  grid-gap: 0.25rem;
  max-width: 50rem;
}
.close {
  margin-left: 1rem;
}
@media(min-width:760px) {
  .icon {
    display: inline-flex;
    font: 1rem 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
    width: calc(6px + 2.125ex);
    height: calc(6px + 2.125ex);
  }
}
"
`;

exports[`getComponentCss() should return correct css for state: info and theme: light 1`] = `
":host {
  display: grid;
  grid-template-columns: 1fr auto;
  grid-template-rows: auto;
  grid-row-gap: 1rem;
  align-items: start;
  justify-items: start;
  padding: 1rem;
  background: #E8EEFD;
  border-radius: 4px;
  box-shadow: 0 0.125rem 0.25rem 0 rgba(0,0,0,0.05),0 0.9375rem 1.25rem 0 rgba(0,0,0,0.2);
}
p {
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  margin: 0;
  color: #010205;
}
@media(min-width:760px) {
  :host {
    grid-template-columns: 2rem 1fr auto auto;
  }
}

.icon {
  display: none;
}
.content {
  display: grid;
  grid-gap: 0.25rem;
  max-width: 50rem;
}
.close {
  margin-left: 1rem;
}
@media(min-width:760px) {
  .icon {
    display: inline-flex;
    font: 1rem 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
    width: calc(6px + 2.125ex);
    height: calc(6px + 2.125ex);
  }
}
"
`;

exports[`getComponentCss() should return correct css for state: neutral and theme: dark 1`] = `
":host {
  display: grid;
  grid-template-columns: 1fr auto;
  grid-template-rows: auto;
  grid-row-gap: 1rem;
  align-items: start;
  justify-items: start;
  padding: 1rem;
  border-radius: 4px;
  box-shadow: 0 0.125rem 0.25rem 0 rgba(0,0,0,0.05),0 0.9375rem 1.25rem 0 rgba(0,0,0,0.2);
}
p {
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  margin: 0;
  color: #FBFCFF;
}
@media(min-width:760px) {
  :host {
    grid-template-columns: 2rem 1fr auto auto;
  }
}

.icon {
  display: none;
}
.content {
  display: grid;
  grid-gap: 0.25rem;
  max-width: 50rem;
}
.close {
  margin-left: 1rem;
}
@media(min-width:760px) {
  .icon {
    display: inline-flex;
    font: 1rem 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
    width: calc(6px + 2.125ex);
    height: calc(6px + 2.125ex);
  }
}
"
`;

exports[`getComponentCss() should return correct css for state: neutral and theme: light 1`] = `
":host {
  display: grid;
  grid-template-columns: 1fr auto;
  grid-template-rows: auto;
  grid-row-gap: 1rem;
  align-items: start;
  justify-items: start;
  padding: 1rem;
  border-radius: 4px;
  box-shadow: 0 0.125rem 0.25rem 0 rgba(0,0,0,0.05),0 0.9375rem 1.25rem 0 rgba(0,0,0,0.2);
}
p {
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  margin: 0;
  color: #010205;
}
@media(min-width:760px) {
  :host {
    grid-template-columns: 2rem 1fr auto auto;
  }
}

.icon {
  display: none;
}
.content {
  display: grid;
  grid-gap: 0.25rem;
  max-width: 50rem;
}
.close {
  margin-left: 1rem;
}
@media(min-width:760px) {
  .icon {
    display: inline-flex;
    font: 1rem 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
    width: calc(6px + 2.125ex);
    height: calc(6px + 2.125ex);
  }
}
"
`;

exports[`getComponentCss() should return correct css for state: success and theme: dark 1`] = `
":host {
  display: grid;
  grid-template-columns: 1fr auto;
  grid-template-rows: auto;
  grid-row-gap: 1rem;
  align-items: start;
  justify-items: start;
  padding: 1rem;
  background: #003320;
  border-radius: 4px;
  box-shadow: 0 0.125rem 0.25rem 0 rgba(0,0,0,0.05),0 0.9375rem 1.25rem 0 rgba(0,0,0,0.2);
}
p {
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  margin: 0;
  color: #FBFCFF;
}
@media(min-width:760px) {
  :host {
    grid-template-columns: 2rem 1fr auto auto;
  }
}

.icon {
  display: none;
}
.content {
  display: grid;
  grid-gap: 0.25rem;
  max-width: 50rem;
}
.close {
  margin-left: 1rem;
}
@media(min-width:760px) {
  .icon {
    display: inline-flex;
    font: 1rem 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
    width: calc(6px + 2.125ex);
    height: calc(6px + 2.125ex);
  }
}
"
`;

exports[`getComponentCss() should return correct css for state: success and theme: light 1`] = `
":host {
  display: grid;
  grid-template-columns: 1fr auto;
  grid-template-rows: auto;
  grid-row-gap: 1rem;
  align-items: start;
  justify-items: start;
  padding: 1rem;
  background: #EBFAF0;
  border-radius: 4px;
  box-shadow: 0 0.125rem 0.25rem 0 rgba(0,0,0,0.05),0 0.9375rem 1.25rem 0 rgba(0,0,0,0.2);
}
p {
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  margin: 0;
  color: #010205;
}
@media(min-width:760px) {
  :host {
    grid-template-columns: 2rem 1fr auto auto;
  }
}

.icon {
  display: none;
}
.content {
  display: grid;
  grid-gap: 0.25rem;
  max-width: 50rem;
}
.close {
  margin-left: 1rem;
}
@media(min-width:760px) {
  .icon {
    display: inline-flex;
    font: 1rem 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
    width: calc(6px + 2.125ex);
    height: calc(6px + 2.125ex);
  }
}
"
`;
